erigon-lib/common/dbg/mem.go

```diff
@@ -0,0 +1,50 @@
+package dbg
+
+import (
+	"github.com/ledgerwatch/erigon-lib/metrics"
+	"github.com/shirou/gopsutil/v3/process"
+	"os"
+)
+
+var (
+	memRssGauge          = metrics.NewGauge(`mem_rss`)
+	memSizeGauge         = metrics.NewGauge(`mem_size`)
+	memPssGauge          = metrics.NewGauge(`mem_pss`)
+	memSharedCleanGauge  = metrics.NewGauge(`mem_shared_clean`)
+	memSharedDirtyGauge  = metrics.NewGauge(`mem_shared_dirty`)
+	memPrivateCleanGauge = metrics.NewGauge(`mem_private_clean`)
+	memPrivateDirtyGauge = metrics.NewGauge(`mem_private_dirty`)
+	memReferencedGauge   = metrics.NewGauge(`mem_referenced`)
+	memAnonymousGauge    = metrics.NewGauge(`mem_anonymous`)
+	memSwapGauge         = metrics.NewGauge(`mem_swap`)
+)
+
+func ReadVirtualMemStats() (process.MemoryMapsStat, error) {
+	pid := os.Getpid()
+	proc, err := process.NewProcess(int32(pid))
+
```

Comment:
let's don't leave empty line between error creation and error handling. it increasing chance that somebody will add code there in future.